src/Reshuffle.ts

```diff
@@ -175,14 +188,20 @@ class HttpMultiplexer {
     this.originalPath = originalPath
     this.delegates = []
   }
-  handle(req: any, res: any, next: any) {
+  async handle(req: any, res: any, next: any) {
     req.originalPath = this.originalPath
-    if (this.delegates.length === 0) {
-      const errorMessage = `No handler registered for ${req.method} ${req.originalPath}`
-      return res.status(501).send(errorMessage)
+    let handled = false
+
+    if (this.delegates.length > 0) {
+      for (const delegate of this.delegates) {
+        if (!handled) {
+          handled = await delegate.handle(req, res, next)
+        }
+      }
+    }
+
+    if (!handled) {
+      next()
     }
-    this.delegates.forEach(async function (delegate) {
```

Comment:
Something I learnt the hard way, the forEach is asynchronous, a for loop with await is synchronous!